### PR TITLE
Fixed duplicated function and enabled Prettier

### PR DESCRIPTION
**Pull request by manuelperdigons**
Hey,

I am committing changes we did on Monday about duplicated function, also fixed issue about eslint and prettier config, please take a look.

I still have pending to do the test for the this project and that would be it by me, if you have any change suggestion or idea in order to add some feature or something would be cool, also some advises for the test

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,36 +1,39 @@
...
-    "no-return-await": "off",
-    "no-underscore-dangle": "off",
-    "class-methods-use-this": "off",
-    "prefer-destructuring": ["error", { "object": true, "array": false }],
-    "no-unused-vars": ["error", { "argsIgnorePattern": "req|res|next|val" }]
+        "prettier/prettier": "error",
+        "spaced-comment": "off",
+        "no-console": "warn",
+        "consistent-return": "off",
+        "func-names": "off",
```
Comment (brahyam) [inline .eslintrc.json:18]: Maybe could be useful to have this one on? What do you think?. check https://eslint.org/docs/rules/func-names

File: controllers/getFlightsData.js [hunk 2]
```diff
@@ -10,11 +10,9 @@ exports.getFlightsAPI1 = async (url, res) => {
                 password: 'supersecret'
             }
         })
-        return session
+        return session.data.flights
```
Comment (brahyam) [inline controllers/getFlightsData.js:13]: What do you think about adding `username` and `password` as env variables?

File: controllers/getFlightsData.js [hunk 3]
```diff
@@ -10,11 +10,9 @@ exports.getFlightsAPI1 = async (url, res) => {
...
     } catch (error) {
-        res.status(400).json({
-            status: 'failed',
-            message: 'An error has happened getting all flights, please try again'
-        })
+        console.error('Unable to get flights, please try again');
```
Comment (brahyam) [inline controllers/getFlightsData.js:15]: As this message is not going to be seen by any user, only developers in the console. could be useful to have more details about the error in the log to be able to debug in the future. eg. including http error number and real error message

File: controllers/index.js [hunk 4]
```diff
@@ -1,59 +1,65 @@
...
+    if (flight1.price != flight2.price) {
+        return false;
+    }
+    if (isSameSlice(flight1.slices, flight2.slices) == false) {
+        return false
+    }
+    return true
+};
+
+exports.deleteDuplicates = async (source1, source2) => {
```
Comment (brahyam) [inline controllers/index.js:33]: Maybe the function name could be adjusted to describe better what is doing as is not only deleting duplicates but merging two lists

File: controllers/index.js [hunk 5]
```diff
@@ -1,59 +1,65 @@
...
+        }
+    }
+    return [...source2, ...flightsToAdd];
 }
 
 exports.response = async (req, res, next) => {
     try {
-        const getRequest = await this.getFlights();
-        const deleteDuplicates = await this.deleteDuplicates();
+        const source1 = await flightsAPI.getFlightsAPI1(process.env.LINK_ONE);
```
Comment (brahyam) [inline controllers/index.js:52]: `getFlightsAPI1` sounds tricky. Is there an `getFlightsAPI2` ..3, 4 ? what would be the difference? Also the API at the end of the method name is redundant as the base class is `flightsAPI`

File: controllers/index.js [hunk 6]
```diff
@@ -1,59 +1,65 @@
...
             status: 'success',
-            data: JSON.stringify(deleteDuplicates)
+            data: JSON.stringify(flights)
         })
-
     } catch (err) {
         res.status(400).json({
             status: 'failed',
-            message: 'An error has happened getting all flights, please try again'
+            message: err.message
```
Comment (brahyam) [inline controllers/index.js:62]: As this is the errors users see, I would suggest to leave the generic one. We could log the real one instead with `console.error` to be able to see it while developing

**Review by brahyam — APPROVED** [review]
Duuuuuuude. Very niceee. Keep going! is almost done!

**Comment by manuelperdigons** [comment]
Ok, I checked all your suggestions and I just added these changes to the project, I will do another pull request tomorrow morning with these changes and some tests I will add to the project to finish it. Thanks for the advises bro

**Comment by brahyam** [comment]
Nooooo, it's better to wait and push the changes to this PR because then we have a way to track the comments and which files changed after the comments. When you open a new one all references stay here and I'll have to check all the code again 💩

**Comment by manuelperdigons** [comment]
Mieeeerda, ok got it. Do I revert it? Or leave it like that and I know it for next PR I do?

**Comment by brahyam** [comment]
Let's open another one and continue there 👍  just for future references

**Comment by manuelperdigons** [comment]
Okk, I am doing the tests, also I will set a timeout to the request and I will do the other pull request and that's it 🙌 🙌

**Pull request merged**